SnapWarShareExtension/SWSharePageViewController.swift

```diff
@@ -0,0 +1,150 @@
+import UIKit
+
+class SWSharePageViewController: UIPageViewController {
+    fileprivate var images: [UIImage]?
+    fileprivate var pageViewManager: SWSharePageViewManager?
+    
+    required init(images: [UIImage]) {
+        super.init(transitionStyle: .scroll, navigationOrientation: .horizontal, options: nil)
+
+        self.images = images
+        self.view.backgroundColor = UIColor.black
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        self.dataSource = self
```

Comment:
@misspia this is what i'm talking about for datasource and delegate. i'm implementing the datasource and delegate as extensions near the bottom of the file. basically just overriding some functions that the page view controller will read to determine how to render the carousel view. effectively, you are just "delegating" the work to some functions you define.